### ISSUE TITLE
Cannot parse invalid message-id

### ISSUE THREAD

**Issue by breedbekkikker** [jstedfast/MailKit#976]
Hi,

While fetching the IMessageSummaries of a folder (with MailKit), a parsing error occurs because of an invalid message-id. 

The stacktrace:

```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at MimeKit.Utils.ParseUtils.TryParseMsgId(Byte[] text, Int32& index, Int32 endIndex, Boolean requireAngleAddr, Boolean throwOnError, String& msgid)
   at MimeKit.Utils.MimeUtils.ParseMessageId(Byte[] buffer, Int32 startIndex, Int32 length)
   at MimeKit.Utils.MimeUtils.ParseMessageId(String text)
   at MailKit.Net.Imap.ImapUtils.<ParseEnvelopeAsync>d__40.MoveNext()
```

The MIME headers:

```
Date: Thu, 26 Dec 2019 08:01:12 +0100 (CET)
From: XYZ <abc@def.ghi>
Reply-To: abc@def.ghi
To: xyz@abc.def
Message-ID: ACME20191226@423549219_ACME20191226@
Subject: ACME
MIME-Version: 1.0
Content-Type: multipart/related; 
	boundary="----=_Part_4389_324343.157734362342342"
```

Is it possible to make a workaround for these kind of invalid message ids? Thanks.

**Comment by jstedfast** [comment]
Done ;-)

**Comment by breedbekkikker** [comment]
Thanks, works like a charm!

**Comment by jstedfast** [comment]
Just released MailKit 2.5.1 with this fix.
